src/test/java/teammates/test/cases/datatransfer/AccountAttributesTest.java

```diff
@@ -133,6 +135,72 @@ public void testLegacyAccountEntityToAttributes() {
 
     }
 
+    @Test
+    public void shouldCreateDeepCopy() {
+        AccountAttributes account = createInvalidAccountAttributesObject();
+        account.isInstructor = true;
+
+        String profileShortName = "John";
+        String profilePersonalEmail = "person@email.com";
+        String profileInstitute = "profile institute test profile 23";
+        String profileNationality = "nationality g35";
+        String profileGender = "male";
+        String profileMoreInfo = "some test info 221";
+        String profilePictureKey = "picture key 223";
+
+        account.studentProfile = StudentProfileAttributes.builder()
+                .withGoogleId(account.googleId)
+                .withShortName(profileShortName)
+                .withEmail(profilePersonalEmail)
+                .withInstitute(profileInstitute)
+                .withNationality(profileNationality)
+                .withGender(profileGender)
+                .withMoreInfo(profileMoreInfo)
+                .withPictureKey(profilePictureKey)
+                .build();
+
+        AccountAttributes copy = account.getCopy();
+
+        assertThat(account, not(copy));
+        assertTrue(account.isInstructor);
+        assertEquals(account.googleId, copy.googleId);
+        assertEquals(account.name, copy.name);
+        assertEquals(account.institute, copy.institute);
+        assertEquals(account.email, copy.email);
+
+        assertThat(account.studentProfile, not(copy.studentProfile));
+        assertEquals(copy.studentProfile.googleId, account.studentProfile.googleId);
+        assertEquals(copy.studentProfile.shortName, profileShortName);
```

Comment:
Although it works, I don't recommend to swap the order of parameters (should be expected value comes first and then actual value for testNG). Especially in this case, the order is not consistent in your comparision.